### PR TITLE
Segmentation Survey: Integrate survey component with entrepreneur flow

### LINKED ISSUE #89550: Integrate the Segmentation Survey component with the entrepreneur flow

### PR DESCRIPTION
**Pull request by phcp**
Closes https://github.com/Automattic/wp-calypso/issues/89550

## Proposed Changes

* Integrate the Segmentation Survey component with the entrepreneur flow
* Remove "e-commerce segmentation flow", which was created for testing purposes
* Update navigation logic to use hash parameters
* Hide back button on the first page

## Testing Instructions

* Apply this PR to your local
* Go to http://calypso.localhost:3000/setup/entrepreneur/start?flags=ecommerce-segmentation-survey
* Check if the survey is properly displayed
* Navigate through the questions and verify if the save request is performed
  * The hash param should be updated in the URL
* After the last question, the step is submitted, and the site created (the flow development is still in progress)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by mashikag** [comment]
Things that I have noticed when testing locally. Keep in mind those are general observations and are not strictly related to this PR, but will need to be addressed either way.
1. The radio button selected looks off.
2. Every time a radio button is selected, an SVG is requested. This should happen once only, at most. 
3. The layout on a small screen off in relation to the header (where Back and Skip buttons are).

https://github.com/Automattic/wp-calypso/assets/2019970/0a233667-817e-49ac-90e2-7dc76038fdcc

**Comment by mashikag** [comment]
Also, I noticed that all components on the page are re-rendered on any option selected. It should not be necessary to do so. In the recording I am printing a log message on every `CustomDocumentHead` re-render.

https://github.com/Automattic/wp-calypso/assets/2019970/1a5cfb74-b30f-418b-837c-68570977ecab

**Comment by yansern** [comment]
Just dropping in to test the navigation part to see if it works both via the UI buttons and the browser back/forward buttons. They work just fine. Thanks!

**Comment by yansern** [comment]
Not for this PR: I think the best time to clear the survey answers is after the completion of the `create-site` step. It could be wedged in between `create-site` and `processing` under `waitForAtomic`, best to add this flow rework PR https://github.com/Automattic/wp-calypso/pull/89627 goes out.

Just writing this here because I see you've introduced `clearAnswers` in this PR.

File: client/components/survey-container/components/style.scss [hunk 1]
```diff
@@ -1,6 +1,7 @@
 @import "@automattic/color-studio/dist/color-variables";
 @import "@automattic/typography/styles/variables";
 @import "@wordpress/base-styles/breakpoints";
+@import "client/components/forms/form-radio/style.scss";
```
Comment (phcp) [inline client/components/survey-container/components/style.scss:4]: @ivan-ottinger I wasn't able to see the radio button after applying the Survey component to the stepper flow. I noticed your radio button is dependent on `FormRadio` styles (which makes sense), so I imported the styles here, and it worked fine. What do you think?

Below is a screenshot of how it looks without this line:
Screenshot 2024-04-17 at 10 32 24

Comment (ivan-ottinger) [reply client/components/survey-container/components/style.scss:4]: Thanks, @phcp. That sounds great. 👍🏼👍🏼

**Comment by phcp** [comment]
Thanks for the reviews! @mashikag @yansern 

> The radio button selected looks off.

This was fixed after the rebase with Ivan's PR.

> Every time a radio button is selected, an SVG is requested. This should happen once only, at most.

It seems to be an issue with the `Gridicon`, as it's a memoized component and should not be re-rendered without parameter changes. It can also possibly be an issue with how the `Gridicon` is applied to the `StepContainer` component (as part of the Stepper components). I tried a few ideas, but nothing worked; it probably needs some investigation and could be worked on another PR. I can create a follow-up task for that.

> The layout on a small screen off in relation to the header (where Back and Skip buttons are).

Fixed on this PR, thanks for catching that.

> Also, I noticed that all components on the page are re-rendered on any option selected. It should not be necessary to do so. In the recording I am printing a log message on every CustomDocumentHead re-render.

Considering that the higher component on the hierarchy is the one holding the answers data, I think it's expected to re-render the children components when the answers change. I memoized the variables that could trigger unnecessary renders, but it's not a complete solution. The only way I see to reduce that would be to change the components structure to couple the data and routing layer with the visual component, possibly removing the provider. What do you think?

**Review by yansern — APPROVED** [review]
Test works fine. Code looks fine. The document head re-rendering on every option select is a trivial issue. Thanks for the `useCachedAnswers` hook.

**Pull request merged**